utils/HtmlStructures.ts

```diff
@@ -0,0 +1,81 @@
+    // Output message Html basis
+    const formalOutput = ( content: string ) => QueryResult( QueryResultContent( content ) )
+    // App's top buttons structure
+    const topButtonsRow: string = ( SearchBtn( clickedBtn ) + " " + AddBtn( clickedBtn ) + " " + RemoveBtn( clickedBtn ) + " " + ShowAllBtn( clickedBtn ) + Space() )
+    // Input field structure
+    const inputField = ( state.btnState !== CurrentButton.ShowAll ) ? InputSubmitElems( "Put your input here, please", clickedBtn ) : ""
+    // According to current state decide what output to show
+    const switchOutput = ( ( state ) => {
+        switch ( clickedBtn ) {
+            case CurrentButton.Search:
+                // Output if at least one user found
+                if ( state.bIsSbmtClicked === true && users.length > 0 ) {
+                    return HiddenDiv( clickedBtn, users, state.bIsSbmtClicked, Table( TheadName( "Name" ), TheadName( "Id" ), TheadName( "Date of birth" ), Tbody( users, originalUsers ) ) )
+                }
+                // Output there are no found users
+                else if ( users.length === 0 ) {
+                    return formalOutput( "No users found!" )
+                }
+                break
+            case CurrentButton.Add:
+                // Show detailed output with info of added user if input is right
+                if ( state.bIsSbmtClicked === true && state.bIsExecutedRight === true ) {
+                    return formalOutput( "Added user: ID: " + users.indexOf( users[users.length - 1] ) + "; Full Name: " + users[users.length - 1].name + " " + users[users.length - 1].surname + "; Born in: " + users[users.length - 1].dateOfBirth + ";" )
+                }
+                // Output if input while adding new user is wrong
+                else if ( state.bIsExecutedRight === false ) {
+                    return formalOutput( "Please, write name, surname, date of birth of user" )
+                }
+                break
+            case CurrentButton.Remove:
+                // Show detailed output with info of removed user if input is right
+                if ( state.bIsSbmtClicked === true && state.bIsExecutedRight === true ) {
+                    return formalOutput( "Removed user: Full Name: " + state.removedUser[0].name + " " + state.removedUser[0].surname + "; Born in: " + state.removedUser[0].dateOfBirth + ";" )
+                }
+                // Output if input while removing user is wrong
+                else if ( state.bIsExecutedRight === false ) {
+                    return formalOutput( "Please, write the existing id of user" )
+                }
+                break
+            case CurrentButton.ShowAll:
+                return HiddenDiv( clickedBtn, users, state.bIsSbmtClicked, Table( TheadName( "Name" ), TheadName( "Id" ), TheadName( "Date of birth" ), Tbody( users, users ) ) )
+        }
+    }
+    )( state )
```

Comment:
Why did you use IIFE here at all? You have IIFE with a switch() statement inside and nothing else. Why not just use switch statement directly, without IIFE?